### PR TITLE
GUVNOR-2173 - Asset Oracle: Change Impact prediction 

### PR DESCRIPTION
**Pull request by mrietveld**
Change Impact, Part I: building a reference infrastructure that we can query.

For an explanation, see https://github.com/droolsjbpm/kie-wb-common/pull/414

PR set:
- https://github.com/uberfire/uberfire/pull/455
- https://github.com/droolsjbpm/kie-wb-common/pull/414
- https://github.com/droolsjbpm/drools-wb/pull/257
- https://github.com/droolsjbpm/jbpm-designer/pull/375
- https://github.com/droolsjbpm/jbpm-form-modeler/pull/63

### CODE REVIEW

File: drools-wb-screens/drools-wb-dsl-text-editor/drools-wb-dsl-text-editor-backend/src/main/java/org/drools/workbench/screens/dsltext/backend/server/indexing/DslFileIndexer.java [hunk 1]
```diff
@@ -33,37 +31,29 @@
...
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.uberfire.backend.server.util.Paths;
-import org.uberfire.ext.metadata.engine.Indexer;
-import org.uberfire.ext.metadata.model.KObject;
-import org.uberfire.ext.metadata.model.KObjectKey;
-import org.uberfire.io.IOService;
 import org.uberfire.java.nio.file.Path;
 
+// TODO: yeah, so there's no "rule => <rule name>" key-value pair added to the lucene index doc here
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dsl-text-editor/drools-wb-dsl-text-editor-backend/src/main/java/org/drools/workbench/screens/dsltext/backend/server/indexing/DslFileIndexer.java]: Yeah.. bit oh a hole in the new indexing :-(

Is the enumerated ```ResourceType``` sufficiently flexible to support all existing (and future) asset types?

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/DecisionTableXLSFileIndexer.java [hunk 2]
```diff
@@ -70,66 +46,30 @@ public boolean supportsPath( final Path path ) {
...
                 } catch ( IOException e ) {
+                    // no-op
                 }
             }
         }
-
-        return index;
     }
 
+    /*
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/main/java/org/drools/workbench/screens/dtablexls/backend/server/indexing/DecisionTableXLSFileIndexer.java]: Really!?!? Adding an empty Javadoc block :-(

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/config/README.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+This file is needed to preserve this directory (config/).
```
Comment (manstis) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/config/README.txt]: I've seen this in a few places.. why the need/introduction of JNDI? Did I miss something somewhere?

**Comment by manstis** [comment]
On the whole, nice rationalisation of the different DRL-based Indexers :-)

File: drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/config/README.txt (hunk 3 above)
Comment (mrietveld) [inline drools-wb-screens/drools-wb-dtable-xls-editor/drools-wb-dtable-xls-editor-backend/src/test/resources/config/README.txt]: I don't think so -- I just know that there's a weird error that shows up (but doesn't affect the test) "normally". So I googled it, found the solution and added this.

**Comment by mrietveld** [comment]
jenkins please retest

**Comment by mrietveld** [comment]
jenkins plese retest

(Fixed errai marshalling issues)

**Comment by manstis** [comment]
@mrietveld This is still red :-(

**Comment by mrietveld** [comment]
Jenkins, if you wouldn't mind, please consder a retest -- it would make me pretty happy. 

(Curious about just how the parsing algorithm for this works.. ;D)

**Comment by mrietveld** [comment]
Are you there, Jenkins? It's me, Marco. Would you please retest?

**Pull request merged**